src/main/scala/io/iohk/praos/crypto/package.scala

```diff
@@ -0,0 +1,51 @@
+package io.iohk.praos
+
+import akka.util.ByteString
+
+package object crypto {
+  type Key = ByteString
+  type KeyPair = (Key, Key)
+  type SecureRandom = Int
+
+  // Dummy implementation to simulate a public-private key schema.
+  val keyLength = 32
+  private def genKey = (s: SecureRandom, length: Int) => ByteString(new Array[Byte](length) map(_ => s.toByte))
+
+  private def XOR(x: ByteString, y: ByteString): ByteString =
+    ByteString((x zip y).map(elements => (elements._1 ^ elements._2).toByte).toArray)
+  
+  // Post: Return (publicKey, privateKey)
+  def generateKeyPair(seed: SecureRandom): KeyPair = {
+    (genKey(seed + 1, keyLength), genKey(seed, keyLength))
+  }
+
+  def getPublicKeyFromPrivateKey(privateKey: Key): Key = {
+    privateKey map(byte => (byte + 1).toByte)
+  }
+
+  trait Cipher {
+    def encryptWith(data: ByteString, publicKey: Key): ByteString
+    def decryptWith(encryptedData: ByteString, privateKey: Key): Option[ByteString]
+  }
+
+  trait Signer {
+    // Post: Return signedData
```

Comment:
Minor detail: Maybe we could replace this comment and other similar in the PR with Scaladoc complaint comments?